src/Atmos/Parameterizations/SurfaceFluxes/SurfaceFluxes.jl

```diff
@@ -231,66 +247,102 @@ end
+
+Computes temperature scale `θ_star` given
+
+ - `Δz` layer thickness (not spatial discretization)
+ - `a` free model parameter with prescribed value of 4.7
+ - `xatol` absolute error tolerance for iteration
+ - `maxiters` maximum number of iterations
+ - `θ_bar` basic potential temperature
+ - `pottemp_flux` potential temperature flux
+ - `z_0` roughness length (for heat)
+ - `x_ave` volume-averaged potential temperature
+ - `x_s` surface potential temperature
+ - `coeff` equation coefficient (1/Prantl)
+ - `u_star` friction velocity
+"""
+function compute_temperature_scale(Δz, a, xatol, maxiters, θ_bar, pottemp_flux, z_0, x_ave, x_s, coeff, u_star)
+  x_star_0 = x_ave * k_Karman / log(Δz / z_0)
+  x_star = x_star_0
+  let x_ave=x_ave, θ_bar=θ_bar, Δz=Δz, z_0=z_0, a=a, xatol=xatol, maxiters=maxiters
+    function compute_x_star(x)
+      L = compute_MO_len(u_star, θ_bar, pottemp_flux) # Note difference with compute_friction_velocity
       R_z0 = compute_R_z0(z_0, Δz)
       temp1 = log(Δz/z_0)
-      temp2 = - Ψ_m_closure(Δz/L, L)
-      temp3 = z_0/Δz * Ψ_m_closure(z_0/L, L)
-      temp4 = R_z0 * (ψ_m_closure(z_0/L, L) - 1)
-      return (temp1+temp2+temp3+temp4) / k_Karman
+      temp2 = - compute_Psi_m(Δz/L, L, a)
+      temp3 = z_0/Δz * compute_Psi_m(z_0/L, L, a)
+      temp4 = R_z0 * (compute_psi_m(z_0/L, L, a) - 1)
+      return coeff*k_Karman/(temp1+temp2+temp3+temp4)*(x_ave-x_s)
     end
-    compute_ustar(u) = u_ave/compute_u_ave_over_ustar(u)
-    ustar_1 = compute_ustar(ustar_0)
-    ustar, converged = RootSolvers.find_zero(
-      u -> u_ave - u*compute_u_ave_over_ustar(u),
-      ustar_0, ustar_1, SecantMethod(),
-      tol_abs, iter_max)
+    compute_roots(x) = x - compute_x_star(x)
+    x_star_1 = compute_x_star(x_star_0)
+    x_star, converged = RootSolvers.find_zero(x -> compute_roots(x), x_star_0, x_star_1, SecantMethod(),
+                                              xatol, maxiters)
   end
-  return ustar
+  return x_star
 end
 
 """
-    compute_exchange_coefficients(z, F_m, F_h, a, u_star, θ, flux, Pr)
+    compute_exchange_coefficients(z, F_m, F_h, a, u_star, θ_star, θ_bar, Pr, pottemp_flux)
 
-Computes exchange transfer coefficients:
+Computes exchange transfer coefficients
 
-  - K_D  momentum exchange coefficient
-  - K_H  thermodynamic exchange coefficient
-  - L_mo Monin-Obukhov length
+  - `K_D`  momentum exchange coefficient
+  - `K_H`  thermodynamic exchange coefficient
+  - `L_MO` Monin-Obukhov length
 """
-function compute_exchange_coefficients(z, F_m, F_h, a, u_star, θ, flux, Pr)
+function compute_exchange_coefficients(z, F_m, F_h, a, u_star, θ_star, θ_bar, Pr, pottemp_flux)
 
-  L_mo = compute_MO_len(u_star, θ, flux)
-  ψ_m = compute_ψ_m(z/L_mo, L_mo, a)
-  ψ_h = compute_ψ_h(z/L_mo, L_mo, a, Pr)
+  L_MO = compute_MO_len(u_star, θ_bar, pottemp_flux)
+  ψ_m = compute_psi_m(z/L_MO, L_MO, a)
+  ψ_h = compute_ψ_h(z/L_MO, L_MO, a, Pr)
```

Comment:
Why Unicode psi here and not in line above?

Reply:
I changed these because it is very difficult to distinguish the upper and lowercase `psi`. I forgot to replace all of them, I'll spell them all out (none should be exposed to the user anyway since the user should only have to call `surface_conditions`).